web/src/stream_settings_data.ts

```diff
@@ -50,7 +50,7 @@ function get_subs_for_settings(subs: StreamSubscription[]): SettingsSubscription
     // delegating, so that we can more efficiently compute subscriber counts
     // (in bulk).  If that plan appears to have been aborted, feel free to
     // inline this.
-    return subs.map((sub) => get_sub_for_settings(sub));
+    return subs.filter((sub) => !sub.deactivated).map((sub) => get_sub_for_settings(sub));
```

Comment:
This line seems out of place in the commit it's part of -- should it be an earlier independent commit for making deactivated streams that we're adding to the metadata not immediately appear in the settings UI?

Reply:
Fixed.